src/main/lrsql/util.clj

```diff
@@ -70,9 +74,13 @@
   :ret instant-spec)
 
 (defn str->time
-  "Parse a string into a java.util.Instant timestamp."
+  "Parse an ISO 8601 timestamp string into a java.util.Instant timestamp. The
+  two parse fns are to support the Z and the +00:00 offset timestamp formats"
   [ts-str]
-  (wrap-parse-fn java-time/instant "timestamp" ts-str))
+  (wrap-parse-fn java-time/instant "timestamp" ts-str
+                 :retry-parse-fn #(-> %
+                                      java-time/offset-date-time
```

Comment:
this extra step is unnecessary in newer versions of Java. This is basically just for 11 support where java.time/instant can't directly parse offsets like `+01:00` etc.

Reply:
You should add that as a code comment then.

Reply:
While we're in this code block, I would've done this differently by implementing a `str->time*` function that looks like this:
```clojure
(defn- str->time* [ts-str]
  (try (java-time/instant ts-str)
       (catch IllegalArgumentException _ ; or whatever java-time/instant thows
         (-> ts-str java-time/offset-date-time java-time/instant))))
```
instead of affecting the entire `wrap-parse-fn` macro.

Reply:
Yes but I kinda loved the idea of one day reusing the macro for something ELSE that needed an alternative exception-triggered backup parser. It was more an experiment. Realized fully it could be a *set* of converters to try before dying but i didnt want to waste more time on it.

Reply:
In my view the updated macro is in an awkward spot right now:
- As you alluded, ideally this should support multiple possible backup functions (e.g. if plan B fails move onto plan C)
- If an exception happened on the backup function, it would not get caught (this is more pertinent in the short term)

Reply:
Yes, i think it would, because it's recursively calling the same macro.

Reply:
Oh good point. You should add a test for that though to make sure that that's what _actually_ happens.

Reply:
Yeah ok I thought about a fail case, that's rational.